### DISCUSSION TITLE
Web browser integration over TCP?

### DISCUSSION THREAD

**Discussion by TommoCrabb** [keepassxreboot/keepassxc#7937] [Ideas]
I recently switching over to using flatpaks and ran into the same problem that everyone else is having with getting keepassxc to integrate with sandboxed web browsers. Having read through a few threads on the issue tracker, I can't help but ask the question: Is there any reason that the browser extension can't communicate with keepassxc over a TCP port on the host machine (eg: 127.0.0.1:9852) instead of using a unix socket? I did a quick search of the issue tracker and the discussion section and couldn't find anybody else suggesting this, so I'm guessing there's an obvious reason why this can't/isn't/won't be done - I'm just wondering what it is? Cheers.

**Comment by droidmonkey** [comment]
That's what we migrated from about 3 years ago. It's simply not secure enough (easily eavesdropped) and requires your password manager to have an open tcp port (a web server essentially). We are mulling over whether we will have to add support for this, at least on Linux, on an opt-in basis. Hoping that flatpak can produce a proper native messaging portal soon.

**Reply by TommoCrabb** [reply]
Thanks for the explanation and the quick reply - much appreciated. Is a native messaging portal something the flatpak devs are working on, or is that just something we're all hoping for at the moment?

**Reply by droidmonkey** [reply]
Working on... maybe? We are definitely not the only app/extension impacted by aggressive browser sandboxing.
